model_analyzer/config/generate/brute_plus_binary_parameter_search_run_config_generator.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Optional, Generator, Dict
+
+from .config_generator_interface import ConfigGeneratorInterface
+
+from model_analyzer.config.generate.brute_run_config_generator import BruteRunConfigGenerator
+from model_analyzer.config.generate.model_variant_name_manager import ModelVariantNameManager
+from model_analyzer.config.generate.perf_analyzer_config_generator import PerfAnalyzerConfigGenerator
+from model_analyzer.config.run.run_config import RunConfig
+from model_analyzer.triton.client.client import TritonClient
+from model_analyzer.device.gpu_device import GPUDevice
+from model_analyzer.config.input.config_command_profile import ConfigCommandProfile, is_request_rate_specified
+from model_analyzer.config.generate.model_profile_spec import ModelProfileSpec
+from model_analyzer.result.result_manager import ResultManager
+from model_analyzer.result.run_config_measurement import RunConfigMeasurement
+from model_analyzer.result.parameter_search import ParameterSearch
+
+from model_analyzer.constants import LOGGER_NAME
+
+from copy import deepcopy
+from math import log2
+
+import logging
+
+logger = logging.getLogger(LOGGER_NAME)
+
+
+class BrutePlusBinaryParameterSearchRunConfigGenerator(ConfigGeneratorInterface
```

Comment:
I'm concerned about how many different generators we have. So many classes means so many places to change if something changes.

Seems like we should be able to abstract it away into something more generic. Something along these lines?

Top level generator:
  - make brute vs quick. 
  - Execute it
  - If binary enabled (or some other class to calculate and spawn binary searches)
  -   Create binary search class and execute it

We're considering adding more binary search cases around throughput inflection points. It would be nice if there was just one place to make that change instead of a bunch of child classes.

If you agree, we can just file a follow-up story to scrub the generator class hierarchy.

Reply:
I'm not following. Lets take this offline.

Reply:
Discussed offline